config/confighttp/confighttp_test.go

```diff
@@ -425,3 +427,68 @@ func TestHttpHeaders(t *testing.T) {
 		})
 	}
 }
+
+func TestOAuth2SettingsIncomplete(t *testing.T) {
+	setting := HTTPClientSettings{
+		Endpoint:   "https://some-server/v1/endpoint",
+		TLSSetting: configtls.TLSClientSetting{},
+		OAuth2ClientCredentials: &configoauth2.OAuth2ClientCredentials{
+			ClientID: "testclientid",
+			TokenURL: fmt.Sprintf("%s/v1/token", "https://some-server/v1/token"),
```

Comment:
This looks odd for two reasons: 

1. you'll end up with `https://some-server/v1/token/v1/token`, won't you?
2. both are fixed strings, you don't need Sprintf

Reply:
Aha! sorry embarrassing copy paste error fixed now.